### PR TITLE
Docs: add another akka-http interop example

### PR DESCRIPTION
**Pull request by alexklibisz**
I've found this pattern useful in a few projects. LMK what you think. Happy to iterate on it.

### CODE REVIEW

**Comment by alexklibisz** [comment]
@raboof Any thoughts here? Happy to iterate to get this merged.

File: docs/src/main/paradox/server/akka-http.md [hunk 1]
```diff
@@ -50,18 +50,157 @@ Java
...
+:  @@snip [LoggingErrorHandlingGreeterServer.java](/plugin-tester-java/src/main/java/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.java) { #implementation }
+
+### Method to log, handle, and recover each RPC 
+
+We define a method that returns a @apidoc[Route$] implementing our logging and error recovery features.
+The method takes three parameters.
+
+1. A function that takes a @apidoc[RequestContext] and returns a service implementation. This gives us the opportunity to use the context in the implementation. If we don't need it, we can just ignore the context and return a fixed implementation.
+2. A function that takes an @apidoc[ActorSystem] and returns a partial function from Throwable to gRPC @apidoc[Trailers].
+3. A function that takes the service implementation and an error handler and returns a request handler (a function from @apidoc[HttpRequest] to a Future of @apidoc[HttpResponse]).
```
Comment (patriknw) [inline docs/src/main/paradox/server/akka-http.md:89]: For Java it's not Future, but probably a CompletionStage?
```
@scala[`Future`]@java[`CompletionStage`]
```

**Review by patriknw — COMMENTED** [review]
looking good, thanks

File: docs/src/main/paradox/server/akka-http.md (hunk 1 above)
Comment (alexklibisz) [reply docs/src/main/paradox/server/akka-http.md]: Thanks, Fixed in [3421183](https://github.com/akka/akka-grpc/pull/1618/commits/342118367c0c32d9119ee3a6a1c9173c21a9bb73)

**Review by johanandren — APPROVED** [review]

**Comment by alexklibisz** [comment]
Can't quite tell if the 2.12 build failed related to these changes, or some other issue?

**Comment by johanandren** [comment]
Looks like the code uses a CompletableFuture method only present in a later version than JDK 8, so that probably needs fixing:

> [error] /home/runner/work/akka-grpc/akka-grpc/plugin-tester-java/src/main/java/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.java:57:1: cannot find symbol
[error]   symbol:   method failedFuture(java.lang.IllegalArgumentException)
[error]   location: class java.util.concurrent.CompletableFuture

File: plugin-tester-java/src/main/java/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  //#implementation
+  private static class Impl extends GreeterServiceImpl {
+    public Impl(Materializer mat) {
+      super(mat);
+    }
+    @Override
+    public CompletionStage<HelloReply> sayHello(HelloRequest in) {
+      if (Character.isLowerCase(in.getName().charAt(0))) {
+        return CompletableFuture.failedFuture(new IllegalArgumentException("Name must be capitalized"));
```
Comment (patriknw) [inline plugin-tester-java/src/main/java/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.java:57]:
```suggestion
        CompletableFuture<HelloReply> reply = new CompletableFuture<>();
        reply.completeExceptionally(new IllegalArgumentException("Name must be capitalized"));
        return reply;
```

**Pull request merged**